osquery/database/database.cpp

```diff
@@ -343,4 +350,51 @@ void dumpDatabase() {
     }
   }
 }
+
+Status updateDatabase() {
+  std::vector<std::string> keys;
+  auto s = scanDatabaseKeys(kQueries, keys);
+  if (!s.ok()) {
+    return Status(1, "Failed to lookup legacy query data from database");
+  }
+
+  for (const auto& key : keys) {
+    std::string value;
+    if (!getDatabaseValue(kQueries, key, value)) {
+      LOG(WARNING) << "Failed to get value from database " << key << ": "
+                   << value;
+      continue;
+    }
+
+    pt::ptree tree;
+    try {
+      std::stringstream ss;
+      ss << value;
+      pt::read_json(ss, tree);
+    } catch (const pt::json_parser::json_parser_error& /* e */) {
+      LOG(WARNING) << "Conversion from ptree to RapidJSON failed for " << key
+                   << ": " << value;
+    }
+
+    auto json = JSON::newArray();
+    for (const auto& t : tree) {
+      std::stringstream ss;
+      pt::write_json(ss, t.second);
+
+      rj::Document row;
```

Comment:
Use ` auto row = json.getObject();` if this does not work because of `Parse` then we need to talk. There's another more complicated way.

Reply:
I am scared about this currently because the `row` as a `rj::Document` will have it's own allocator.

Reply:
Actually, this works, nvm.

Reply:
👍 was just going to ask.